### PR TITLE
Add Type to DBStatus, Remove DB & DbServer Id, OOB Network Profile fix

### PR DESCRIPTION
**Pull request by akshmish**
**What this PR does / why we need it**:

1. Adds Type to Database Status 
2. Remove DB & DbServer Id from Database Status
3. OOB Network Profile to be made mandatory - also fix the existing UTs as network profile is now a mandatory input

**How Has This Been Tested?**:
WIP

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE

```

### CODE REVIEW

File: config/crd/bases/ndb.nutanix.com_databases.yaml [hunk 1]
```diff
@@ -25,11 +25,8 @@ spec:
     - jsonPath: .status.status
       name: Status
       type: string
-    - jsonPath: .status.id
-      name: Database Instance ID
-      type: string
-    - jsonPath: .status.dbServerId
-      name: Database Server ID
+    - jsonPath: .status.type
+      name: Type
```
Comment (manavrajvanshi) [inline config/crd/bases/ndb.nutanix.com_databases.yaml:29]: Were these changes made by `make generate` automatically?

Comment (akshmish) [reply config/crd/bases/ndb.nutanix.com_databases.yaml:29]: yes, "make manifests" to be precise.

**Review by manavrajvanshi — APPROVED** [review]
LGTM, thanks!

File: api/v1alpha1/ndb_api_helpers.go [hunk 2]
```diff
@@ -173,16 +173,19 @@ func (inputProfile *Profile) Resolve(ctx context.Context, allProfiles []ProfileR
...
+		}
 	}
 
-	if err != nil {
-		log.Error(err, "could not resolve profile by id or name", "profile type", pType, "id", id, "name", name)
-		return ProfileResponse{}, fmt.Errorf("could not resolve profile by id=%v or name=%v", id, name)
+	if isIdProvided {
+		profileById, err = util.FindFirst(allProfiles, func(p ProfileResponse) bool { return p.Type == pType && p.Id == id })
+		if err != nil {
+			log.Error(err, "could not resolve profile by name", "profile type", pType, "id", id)
```
Comment (yash-ntnx) [inline api/v1alpha1/ndb_api_helpers.go:186]: fix error msg - "profile by id"

**Review by yash-ntnx — CHANGES_REQUESTED** [review]

**Review by yash-ntnx — APPROVED** [review]

**Review by manavrajvanshi — APPROVED** [review]

**Pull request merged**